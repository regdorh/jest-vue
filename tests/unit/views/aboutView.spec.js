import { mount, shallowMount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'

import AboutView from '@/views/AboutView.vue'

describe('AboutView', () => { 
    test('Probando la existencia del componente o vista AboutView ', async () => {
        const router = createRouter({
            history: createWebHistory(),
            routes: [{
                path: '/about',
                name: 'AboutViewVue',
                component: AboutView
            }],
        })
        router.push('/about')
        await router.isReady()

        const wrapper = mount(AboutView, {
            global: {
                plugins: [router]
            }
        })
        expect(wrapper.findComponent(AboutView).exists()).toBe(true)
    })  

    test('ValidaciÃ³n de match con el snapshot', () => {
        const wrapper = shallowMount(AboutView)

        expect( wrapper.html() ).toMatchSnapshot()
    })
 })