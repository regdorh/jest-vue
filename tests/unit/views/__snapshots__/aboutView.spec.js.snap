exports[`AboutView Validaci√≥n de match con el snapshot 1`] = `<h1>About View</h1>`;